lib/ring_buffer/ring_buffer.c

```diff
@@ -79,16 +80,123 @@ int sys_ring_buf_get(struct ring_buf *buf, u16_t *type, u8_t *value,
 	if (likely(buf->mask)) {
 		for (i = 0; i < header->length; ++i) {
 			index = (i + buf->head + 1) & buf->mask;
-			data[i] = buf->buf[index];
+			data[i] = buf->buf.buf32[index];
 		}
 		buf->head = (buf->head + header->length + 1) & buf->mask;
 	} else {
 		for (i = 0; i < header->length; ++i) {
 			index = (i + buf->head + 1) % buf->size;
-			data[i] = buf->buf[index];
+			data[i] = buf->buf.buf32[index];
 		}
 		buf->head = (buf->head + header->length + 1) % buf->size;
 	}
 
 	return 0;
 }
+
+u32_t sys_ring_buf_bytes_alloc(struct ring_buf *buf, u8_t **data, u32_t size)
+{
+	u32_t space, trail_size, allocated;
+
+	space = sys_ring_buf_custom_space_get(buf->size, buf->head,
+					      buf->misc.byte_mode.tmp_tail);
+
+	/* Limit requested size to available size. */
+	size = size > space ? space : size;
+	trail_size = buf->size - buf->misc.byte_mode.tmp_tail;
+
+	/* Limit allocated size to trail size. */
+	allocated = (trail_size > size) ? size : trail_size;
+
+	*data = &buf->buf.buf8[buf->misc.byte_mode.tmp_tail];
+	buf->misc.byte_mode.tmp_tail =
+			(buf->misc.byte_mode.tmp_tail + allocated) % buf->size;
+
+	return allocated;
+}
+
+int sys_ring_buf_bytes_put(struct ring_buf *buf, u32_t size)
+{
+	if (size > sys_ring_buf_space_get(buf)) {
+		return -EINVAL;
+	}
+
+	buf->tail = (buf->tail + size) % buf->size;
```

Comment:
well, yes but it's done one per chunk and not on every byte. Would you prefer something like:
`buf->tail = (buf->tail + size);`
`buf->tail = buf->tail > buf->size ? buf->tail - buf->size : buf->tail;`

Reply:
fixed

Reply:
@nordic-krch :

Sorry, I guess I should have mentioned I'm packing for travel/will travel, so may be slow with responses.

> Would you prefer something like:
> buf->tail = buf->tail > buf->size ? buf->tail - buf->size : buf->tail;

Yes, that's what I hinted at. You see, it was easier with your previous version, which worked char by char ;-). Let me have a look at what you have now.

Reply:
> well, yes but it's done one per chunk and not on every byte.

As I mentioned, my usecase is console subsys, which works byte by byte, so would be affected. (Yes, it might be optimized, but that's a separate task, and a some (much of?) hardware still have byte by byte IRQ).